Hashtable.c

```diff
@@ -120,19 +120,23 @@ Hashtable* Hashtable_new(size_t size, bool owner) {
 void Hashtable_delete(Hashtable* this) {
    Hashtable_clear(this);
 
-   free(this->buckets);
-   free(this);
+   if (this) {
+      free(this->buckets);
+      free(this);
+   }
 }
 
 void Hashtable_clear(Hashtable* this) {
    assert(Hashtable_isConsistent(this));
 
-   if (this->owner)
+   if (this && this->owner)
       for (size_t i = 0; i < this->size; i++)
          free(this->buckets[i].value);
 
-   memset(this->buckets, 0, this->size * sizeof(HashtableItem));
-   this->items = 0;
+   if (this) {
+      memset(this->buckets, 0, this->size * sizeof(HashtableItem));
+      this->items = 0;
+   }
 
    assert(Hashtable_isConsistent(this));
 }
```

Comment:
Please explain these changes. Hashtable functions are always assumed to be called on live objects, thus `this!=NULL` …

Reply:
@BenBE I got errors from GitHub CI "error: null pointer dereference" [here](https://github.com/natoscott/htop/runs/3179357083)
So I added this last commit to trying to fix that

Reply:
Yep. Indicating that you at one place call it without a proper NULL check ;-)

Reply:
@BenBE I will revert this last commit now to see what will happen. thank you :)